sdks/python/apache_beam/pipeline.py

```diff
@@ -525,14 +525,31 @@ def run(self, test_runner_api='AUTO'):
     self.contains_external_transforms = (
         ExternalTransformFinder.contains_external_transforms(self))
 
-    self.contains_run_inference_transform = (
-        RunInferenceSideInputFinder.contains_run_inference_transform(self))
+    # Finds if RunInference has side inputs enables.
+    # also, checks for the side input window is global and has non default
+    # triggers.
+    run_inference_visitor = RunInferenceVisitor().visit_run_inference(self)
+    self._run_inference_contains_side_input = (
+        run_inference_visitor.contains_run_inference_side_inputs)
 
-    if (self.contains_run_inference_transform and
+    self.run_inference_global_window_non_default_trigger = (
+        run_inference_visitor.contains_global_windows_non_default_trigger)
+
+    if (self._run_inference_contains_side_input and
         not self._options.view_as(StandardOptions).streaming):
       raise RuntimeError(
           "SideInputs to RunInference PTransform is only supported "
-          "in streaming mode.")
+          "in streaming mode. To run in streaming mode, add the"
+          " --streaming pipeline option")
+
+    if (self._run_inference_contains_side_input and
+        not self.run_inference_global_window_non_default_trigger):
+      raise RuntimeError(
+          "The RunInference PTransform's SideInput is either using "
+          "GlobalWindows with a default trigger or no Windowing, which "
```

Comment:
```suggestion
          "GlobalWindows with a default trigger, non-global windowing, or no Windowing, which "
```

(probably needs formatting + corresponding test changes)